src/rust/ensogl/lib/text/src/component/area.rs

```diff
@@ -569,7 +569,16 @@ impl Area {
             eval_ sels_cut (m.buffer.frp.delete_left());
 
             eval_ input.paste (m.paste());
-            eval input.paste_string ((s) m.buffer.frp.paste(m.decode_paste(s)));
+            eval input.paste_string([m](s) {
+                let mut fragments = m.decode_paste(s);
+                if m.single_line.get() {
+                    fragments = fragments
+                        .iter()
+                        .map(|fragment| fragment.lines().next().unwrap_or("").to_string())
+                        .collect();
+                }
```

Comment:
Few things:

1. I'm not sure the logic is correct. What are "Fragments"? You are iterating over fragments and for each fragment you are taking the first line. So there may still be multiple lines there from different fragments?
2. This code is really hard to read. Of course its easy when you know what it is doing, but the code should be structured in such a way, that when reading the logic, you should look at a code fragment and after 1-2 seconds you should know what it is doing. Here, you should refactro this logic to a function that is called `dropAllButFirstLine` and use it instead.
3. According to style guide this chained multiline expression should be divided into multiple, well named variables that describe the steps used there. But also, just FYI, you don't have to use the full name `fragment` in variable name, you can use single letter there, as `fragments.iter(|f| ...)` is very easy to read and very understandable.

Reply:
1. I renamed `fragments` to `chunks`, which seems to be consistent with other parts of the code. I assume you get multiple chunks when you copy text from a table and Enso would only use all of them if you have multiple cursors. Is that right?
2. & 3. I refactored it.

Reply:
1. So how does it work when pasting fro table with multiline texts?

Reply:
Before this PR, it pasted one full table row per cursor, including all cells of that row and all possible new-line characters inside those cells.

Now it still pastes one row per cursor, but truncates each row at its first line-break character, which can appear in any cell or not at all. It will still paste multiple cells per row though.

Reply:
Oh, this is per-cursor! That is what I was asking about. Wuld you be so nice and add there docs abut that, so no one will forget how ti works in the future, please?